cluster-autoscaler/core/scaledown/unneeded/nodes.go

```diff
@@ -209,45 +219,56 @@ func (n *Nodes) RemovableAt(autoscalingCtx *ca_context.AutoscalingContext, scale
 	return
 }
 
+// refreshRemovalThreshold calculates the unneeded/unready time for a node and updates the node struct.
+func (n *Nodes) refreshRemovalThreshold(v *node, cp cloudprovider.CloudProvider) {
+	nodeGroup, err := cp.NodeGroupForNode(v.ntbr.Node)
+	if err != nil {
+		klog.Warningf("Error determining node group for %s: %v", v.ntbr.Node.Name, err)
+		return
+	}
+	if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
+		klog.V(4).Infof("Node %s has no node group config", v.ntbr.Node.Name)
+		return
+	}
+
+	readiness, _ := kube_util.GetNodeReadiness(v.ntbr.Node)
+	var removalThreshold time.Duration
+
+	if readiness.Ready {
+		removalThreshold, err = n.sdtg.GetScaleDownUnneededTime(nodeGroup)
+	} else {
+		removalThreshold, err = n.sdtg.GetScaleDownUnreadyTime(nodeGroup)
+	}
+
+	if err != nil {
+		klog.Warningf("Failed to get scale down removalThreshold for %s: %v", v.ntbr.Node.Name, err)
```

Comment:
"Failed to get scale down unneeded/unready time for"? I think "unneeded"/"unready" is going to be familiar for anyone setting these command line flags, while "removalThreshold" is a new concept.